### ISSUE TITLE
Support for Windows containers using `SHELL ["powershell"]`?

### ISSUE THREAD

**Issue by petemounce** [hadolint/hadolint#431]
First of all, please let me say I deeply appreciate the time and effort you have put into this excellent tool; I use it for all my real-computer containers. I just tried it on something a bit different, a Windows container. I'm not really sure whether this is a bug report or a feature request; I vaguely understand that hadolint depends on shellcheck. 

(I don't think it's particularly reasonable of me, a complete stranger, to expect support for a whole new shell as a drive-by issue!)

- [x] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one
    - I found https://github.com/hadolint/hadolint/issues/246 which looks like it works by noticing `SHELL ["pwsh"]`.

### Expected behavior

Should understand that powershell is a _completely reasonable language_ to use within a dockerfile and the world should recognise its brilliance (or, maybe, warn that I'm crazy?).

I'd be totally happy with the same behaviour as #246 does for `pwsh`.

### Actual behavior

Does not recognise my SHELL directive and cannot really figure out the rest, I think.

### Steps to reproduce the behavior

```
$ hadolint workflow/dockers/go-build.img/Dockerfile.windows
workflow/dockers/go-build.img/Dockerfile.windows:5 SC1008 This shebang was unrecognized. ShellCheck only supports sh/bash/dash/ksh. Add a 'shell' directive to specify.
workflow/dockers/go-build.img/Dockerfile.windows:5 SC1073 Couldn't parse this function. Fix to allow more checks.
workflow/dockers/go-build.img/Dockerfile.windows:5 SC1065 Trying to declare parameters? Don't. Use () and refer to params as $1, $2..
workflow/dockers/go-build.img/Dockerfile.windows:5 SC1064 Expected a { to open the function definition.
workflow/dockers/go-build.img/Dockerfile.windows:5 SC1072  Fix any mentioned problems and try again.
workflow/dockers/go-build.img/Dockerfile.windows:9 SC1008 This shebang was unrecognized. ShellCheck only supports sh/bash/dash/ksh. Add a 'shell' directive to specify.
workflow/dockers/go-build.img/Dockerfile.windows:9 SC2239 Ensure the shebang uses an absolute path to the interpreter.
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
$ hadolint --version
Haskell Dockerfile Linter v1.17.3-1-g3fab409
```

**Dockerfile** (if relevant)

```Dockerfile
FROM golang:1.14-windowsservercore-1809

SHELL ["C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe", "-noprofile", "-noninteractive", "-command"]
# Install chocolatey, but turn off the it-slows-things-down download progress feature.
RUN Invoke-Expression ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1')); \
    choco feature disable --name showDownloadProgress

COPY "chocolatey-packages.config" "chocolatey-packages.config"
RUN choco install --no-progress --yes "chocolatey-packages.config"

ENTRYPOINT [ "C:\\Program Files\\Git\\bin\\bash.exe" ]
```

**Additional environment details (OS, stack version, etc.)**

Same outcome with 
* `SHELL ["powershell.exe", "-noprofile", "-noninteractive", "-command"]`
* `SHELL ["powershell", "-noprofile", "-noninteractive", "-command"]`

If I use `SHELL ["pwsh", "-noprofile", "-noninteractive", "-command"]` then I get no output and exit 0 as expected from #246 - but unfortunately my container is based off `windowsservercore-1809`, and has no `pwsh` :/

**Comment by darkvertex** [comment]
I get a lot of SSC2239 with my Windows container using Command Prompt shell.

I declare my shell like so:
```Dockerfile
SHELL ["cmd", "/S", "/C"]
```

Then all my RUN lines pretty much report SC2239 ("_Ensure the shebang uses an absolute path..._") for example:
```Dockerfile
RUN C:\ProgramData\chocolatey\lib\mingw\tools\install\mingw64\bin\mingw32-make.exe -d
```

I guess somewhere there's some regex tuned for Unix commands that is not happy with Windows paths. :man_shrugging:

**Comment by j00ris** [comment]
Related issue: 

Because the backslash is also a file path separator in Windows, using it to denote multi-line statements can cause problems. So [Microsoft recommends](https://docs.microsoft.com/en-us/virtualization/windowscontainers/manage-docker/manage-windows-dockerfile#escape-character) using the `` # escape=` `` parser directive, so replace these backslashes by backticks. 

Hadolint then throws an error on the second line of the multi-line statement: 
_"unexpected '-' expecting '#', '\', ADD, ARG, CMD, COPY, ENTRYPOINT, ENV, EXPOSE, FROM, HEALTHCHECK, LABEL, MAINTAINER, ONBUILD, RUN, SHELL, STOPSIGNAL, USER, VOLUME, WORKDIR, at least one space, or end of input"_

**Comment by okainov** [comment]
As #371 was fixed, now one can use Hadolint on real Windows images, but I see this error now as well...
